presto-main/src/main/java/com/facebook/presto/testing/MaterializedRow.java

```diff
@@ -142,21 +142,45 @@ public int hashCode()
 
     private static class ApproximateDouble
     {
-        private final Double value;
-        private final Double normalizedValue;
+        private final Number value;
+        private final Number normalizedValue;
 
-        private ApproximateDouble(Double value, int precision)
+        private ApproximateDouble(Number value, int precision)
         {
+            checkArgument(value instanceof Float || value instanceof Double, "Only floating point values accepted");
             this.value = value;
+            if (value instanceof Float) {
+                normalizedValue = normalizeFloat((Float) value, precision);
+            }
+            else if (value instanceof Double) {
```

Comment:
It's not clear to me that this is a better approach than adding an ApproximateFloat class. If you stick with this, I suggest renaming the class to be ApproximateNumeric (or whatever the SQL spec calls this class of number).